### PR TITLE
Several cleanups and migrate to Meza

### PR DESCRIPTION
**Pull request by bphenriques**
# Description

See each commit for more details.

# Test Plan
- Authentication ✅ 
- Automatic deploy ✅ 
- Run automated tests in the `stg` environment and validate that the tests are successful ✅ 

For reference:
1. `POST {{host}}/test_case`
```
{
    "id": "f81bbfe2-9d75-4edb-b151-23d84dddea20",
    "name": "Bruno attempt",
    "description": "STG - Test Campaign",
    "created_at": "2020-12-04T15:25:37.528356Z",
    "campaign_templates": {
        "campaign_1": {
            "dialing_mode": "PREDICTIVE",
            "dialing_strategy": {
                "caller_ids": [
                    {
                        "number_id": "5fa9307f8c8a1c0007ff697e",
                        "phone_number": "+18443361654"
                    }
                ],
                "max_attempts": 3,
                "retry_period": 60,
                "retry_time_unit": "SECONDS",
                "calling_hours": [
                    {
                        "from": "09:00:00",
                        "to": "20:00:00",
                        "week_days": [
                            "FRIDAY",
                            "THURSDAY",
                            "MONDAY",
                            "TUESDAY",
                            "WEDNESDAY"
                        ]
                    }
                ],
                "dispositions": {
                    "BUSY": {
                        "status_after_call": "FINAL",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    },
                    "NO_ANSWER": {
                        "status_after_call": "RETRY",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    },
                    "ABANDONMENT": {
                        "status_after_call": "FINAL",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    },
                    "HANGUP_BEFORE_CONNECTION": {
                        "status_after_call": "RETRY",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    },
                    "ANSWERING_MACHINE_DETECTION": {
                        "status_after_call": "FINAL",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    },
                    "INVALID_NUMBER": {
                        "status_after_call": "FINAL",
                        "custom": true,
                        "max_attempts": 2,
                        "retry_period": 2,
                        "retry_time_unit": "SECONDS"
                    }
                }
            },
            "agents": [
                {
                    "agent_external_id": "5fa92b02c4e8d200015ed954"
                }
            ],
            "predictive_mode": {
                "answering_machine_detection": false,
                "max_abandonment_rate": 100.0,
                "max_dialing_ratio": 3,
                "max_ring_time_in_secs": 6,
                "abandonment_timeout": {
                    "amount": 30
                }
            },
            "record_lists": [
                "fe97180e-718a-4cce-b517-5922d8be6ad0"
            ]
        }
    },
    "campaign_ids": [],
    "running_campaign_ids": [],
    "status": "STOPPED"
}
```

2. `POST {{host}}/test_case/f81bbfe2-9d75-4edb-b151-23d84dddea20/actions/run`

Wait a bit for the record lists to be exhausted and then:

Campaign: https://po-automated-tests.gettalkdesk.com/atlas/apps/outbound-dialer/campaigns/2075
Campaign Name: `campaign_1-1e1c8db5-bea7-4adf-a838-31f0a6331247`
Reporting Screenshot (before reducing the call duration from 30s to 15s): 

image

# Deployment Instructions

Mirror `stg` Meza environment variables in the QA environment. We have already the required credentials.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,66 @@
...
+  2. Point the webhook URL to the application's Telnyx webhook endpoint.
+  3. Set the API's version to `API V2`.
+  4. Under `Inbound Settings` set the `SIP subdomain` to `talkdesk.<env>.testing.bot`. Note that this sip-domain is unique and
+     different from the one set in step 1.
+3. Per contact required:
+  1. Buy one in Telnyx.
+  2. Set the phone-number's App to the one created under step 2.
+
+**Notes**:
+* The contact's can't be reused between enviornments.
```
Comment (rubenmfl) [inline README.md:50]: 💅  enviornments

**Review by rubenmfl — APPROVED** [review]
👍  LGTM 👍

**Review by MafaldaSVTd — APPROVED** [review]
🎉

**Comment by MafaldaSVTd** [comment]
Klint is broken...

**Comment by bphenriques** [comment]
> Ktlint is broken...

Checked offline. @MafaldaSVTd  is using a `ktlint` detached from the project rendering different results. Suggested to use the one set up in the project through `./gradlew ktlintCheck`

**Pull request merged**